lib/matplotlib/axes/_axes.py

```diff
@@ -4942,6 +4946,24 @@ def reduce_C_function(C: array) -> float
         tx = x
         ty = y
 
+        xscale_kwargs = {}
+        if isinstance(xscale, tuple):
+            xscale, xscale_kwargs_ = xscale
+            xscale_kwargs = xscale_kwargs_.copy()
+
+        xscale_kwargs.setdefault('base', 10)
+        xscale_kwargs.setdefault('linthresh', 2)
+        xscale_kwargs.setdefault('linscale', 1)
```

Comment:
I'd build the logic the other way round: Create defaults and update them is IMHO more straight forward than filling in non-existing values for defaults.

```suggestion
        symmlog_x_kwargs = {'base': 10, 'linthresh': 2, 'linscale': 2}
        if isinstance(xscale, tuple):
            xscale, kwargs = xscale
            symmlog_x_kwargs.update(kwargs)
```
Note also that I've renamed `xscale_kwargs` to `symmlog_x_kwargs`. These defaults work only on - and are applied only to - the SymmetricalLogTransform. We can make that clear in the name.

Likewise for y.

Reply:
Why would you only apply `xscale_kwargs` to the `SymmetricalLogTransform`? You also need to pass them to the plot scale `self.set_xscale(xscale, **xscale_kwargs)`